web3/_utils/filters.py

```diff
@@ -233,6 +264,48 @@ def is_valid_entry(self, entry: LogReceipt) -> bool:
         return bool(self.data_filter_set_function(entry["data"]))
 
 
+class AsyncLogFilter(AsyncFilter):
```

Comment:
Since this is a direct copy/paste from the `LogFilter` class, I wonder if we can get condense it into one class? I see the inheritance there for `AsyncFilter`. The other thing I was thinking that I'm not sure will work is to have an `is_async` flag (or something) on the class that calls the appropriate async or sync method, depending on the flag. We can also figure that out later in this asyncify-filters process

Reply:
I originally tried to compress them into one (like `class LogFilter(Filter, AsyncFilter) and fell down a mypy wormhole. Maybe having a `BaseLogFilter`? Will try

Reply:
As discussed, no immediate solutions present themselves. Will keep it in mind as asyncing filters moves forward